File: 2021/day1/main.py
def main():
    depths = getInput()
    print(getIncrCount(depths))
    print(slidingWindow(depths))


if __name__ == '__main__':
    main()


def getInput():
    inputs = []
    with open(r"D:\Repositories\AdventOfCode\2021\day1\input.txt", "r") as f:
        for line in f:
            inputs.append(int(line.strip()))
    return inputs


def getIncrCount(arr):
    incr_count = 0
    for i in range(1, len(arr)):
        if(arr[i] > arr[i-1]):
            incr_count += 1
    return incr_count


def slidingWindow(arr):
    window_sums = []
    rem = len(arr) % 3
    for i in range(0, len(arr)-rem):
        window_sums.append(sum(arr[i:i+3]))
    print(getIncrCount(window_sums))

File: 2021/day1/main.rb
def read_input
   File.read("input.txt").split.map(&:to_i)
end

def part_one(arr)
   inc_cnt = 0
   (1..arr.length-1).each do |i|
      if arr[i] > arr[i-1]
         inc_cnt+=1
      end   
   end
   inc_cnt
end

def part_two(arr)
   window_sums = []
   rem = arr.length % 3
   (0..arr.length-rem - 1).each do |i|
      window_sums << arr[i] + arr[i+1] + arr[i+2]
   end
   part_one(window_sums)
end

puts part_one read_input
puts part_two read_input